prometheus/client.go

```diff
@@ -320,6 +320,39 @@ func (in *Client) GetMetricsForLabels(metricNames []string, labelQueryString str
 	return metricsWeFound, nil
 }
 
+// GetExistingMetricNames returns a list of the requested metric names that exist in Prometheus (meaning there is a matching __name__ label).
+func (in *Client) GetExistingMetricNames(metricNames []string) ([]string, error) {
+	if len(metricNames) == 0 {
+		return []string{}, nil
+	}
+
+	log.Tracef("[Prom] GetExistingMetricNames: metricNames=[%v]", metricNames)
+	startT := time.Now()
+	results, warnings, err := in.api.LabelValues(in.ctx, "__name__", []string{}, time.Unix(0, 0), time.Now())
+	if warnings != nil && len(warnings) > 0 {
+		log.Warningf("GetExistingMetricNames. Prometheus Warnings: [%s]", strings.Join(warnings, ","))
+	}
+	if err != nil {
+		return nil, errors.NewServiceUnavailable(err.Error())
+	}
+
+	metricsWeAreLookingFor := make(map[string]bool, len(metricNames))
+	for i := 0; i < len(metricNames); i++ {
+		metricsWeAreLookingFor[string(metricNames[i])] = true
+	}
+
+	metricsWeFound := make([]string, 0, 5)
```

Comment:
I would set the size of this array to the same as the total # we are looking for. Under normal circumstances we are going to get everything we are looking for (only if someone pruned prometheus will it be different, correct?) ... in this case, I would avoid having to reallocate and copy the array as it grows - just set it to the maximum size we expect it could be.
```suggestion
	metricsWeFound := make([]string, 0, len(metricNames))
```